cmd/kubeadm/app/phases/upgrade/staticpods_test.go

```diff
@@ -433,8 +433,8 @@ func TestStaticPodControlPlane(t *testing.T) {
 			if err != nil {
 				t.Fatalf("couldn't run NewFakeStaticPodPathManager: %v", err)
 			}
-			defer os.RemoveAll(pathMgr.(*fakeStaticPodPathManager).KubernetesDir())
-			constants.KubernetesDir = pathMgr.(*fakeStaticPodPathManager).KubernetesDir()
+			tmpKubernetesDir := pathMgr.(*fakeStaticPodPathManager).KubernetesDir()
```

Comment:
`StaticPodPathManager` should have a `KubernetesDir()` method. We should then be able to initialize a path manager with a different KubernetesDir for tests.

Reply:
@rosti Sorry for the delay. Thanks for your suggestion. I'll try it in this direction.